lib/discover/local.go

```diff
@@ -197,6 +197,21 @@ func (c *localClient) registerDevice(src net.Addr, device Device) bool {
 		}
 
 		if len(tcpAddr.IP) == 0 || tcpAddr.IP.IsUnspecified() {
+			srcAddr, err := net.ResolveTCPAddr("tcp", src.String())
```

Comment:
It wasn't as simple as I'd assumed. Here's what I would suggest, typed directly into the browser so may require some fixing:

```go
var srcIP net.IP
if addr, ok := src.(*net.UDPAddr); ok {
    srcIP = addr.IP
} else {
   // Use ResolveTCPAddr or ResolveUDPAddr as you're doing
}
```